core/new-gui/src/app/workspace/types/shared-editing.interface.ts

```diff
@@ -48,7 +48,7 @@ export function createYTypeFromObject<T extends object>(obj: T): YType<T> {
         const yArray = new Y.Array();
         // Create YType for each array item and push
         for (const item of obj as any) {
-          if (item) yArray.push([createYTypeFromObject(item) as unknown]);
+          if (item !== undefined) yArray.push([createYTypeFromObject(item) as unknown]);
```

Comment:
could you elaborate the reason of the change (better in code comments) so that we have context to review?

Reply:
This was a bad practice about `undefined` type equivalence and caused problems in the PR,  so I changed it here.